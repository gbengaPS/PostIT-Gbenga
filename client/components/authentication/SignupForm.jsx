import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';
import Proptypes from 'prop-types';
import Nav from '../common/Nav';
import Container from '../common/Container';
import TextInput from '../common/TextInput';

const SignupForm = (props) => {
  let signupValue = '';
  if (props.loading) {
    signupValue = 'Signing up...';
  } else {
    signupValue = 'Signup';
  }
  return (
    <div>
      <Nav />
      <Container>
        <div className="col s12 m6 offset-m3">
          <div className="form-container">
            <p className="center grey-text big" >Signup</p>
            <p className="red-text">&nbsp; {props.error}</p>
            <form method="POST" action="#" onSubmit={props.handleSubmit}>
              <TextInput
              name="fullName"
              description="Full name"
              handleChange={props.handleChange}
              classnames={classnames('', { invalid: props.errors.fullName })}
              />
              <span className="red-text">&nbsp;{props.errors.fullName}</span>
              <TextInput
              name="username"
              description="Username"
              handleChange={props.handleChange}
              classnames={classnames('', { invalid: props.errors.username })}
              />
              <span className="red-text">&nbsp;{props.errors.username}</span>
              <TextInput
              name="email"
              description="Email"
              handleChange={props.handleChange}
              classnames={classnames('', { invalid: props.errors.email })}
              />
              <span className="red-text">&nbsp;{props.errors.email}</span>
              <TextInput
              name="phoneNumber"
              description="Phone Number"
              handleChange={props.handleChange}
              errorClass="valid"
              classnames={classnames('', { invalid: props.errors.phoneNumber })}
              />
              <span className="red-text">&nbsp;{props.errors.phoneNumber}</span>
              <div className="input-field">
                <input type="password" name="password" id="password"
                onChange={props.handleChange}
                className={classnames('input',
                { invalid: props.errors.password })}
                required />
                <label htmlFor="password"> Password </label>
              </div>
              <span className="red-text" >&nbsp;{props.errors.password}</span>
              <p className="row">
                <input
                type="submit" value={signupValue}
                className=
                "btn align-center light-blue darken-4 col s8 offset-s2"
                />
              </p>
              <p className="center"> Already have and acount?
                <Link to="login"> Login </Link>
              </p>
              <p></p>
            </form>
        </div>
        </div>
      </Container>
    </div>
  );
};
SignupForm.propTypes = {
  errors: Proptypes.object,
  handleChange: Proptypes.func.isRequired,
  handleSubmit: Proptypes.func.isRequired,
  error: Proptypes.string,
  loading: Proptypes.bool,
};
export default SignupForm;
