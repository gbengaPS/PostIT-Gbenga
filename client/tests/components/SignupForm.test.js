import React from 'react';
import 'mock-local-storage';

import SignupForm from '../../components/authentication/SignupForm';

describe('SignupForm component ', () => {
  it('should render correctly', () => {
    const props = {
      errors: {},
      handleChange: () => {},
      handleSubmit: () => {}
    };
    expect(shallow(
    <SignupForm {...props} />)).toMatchSnapshot();
  });
  it('should accept isLoading props', () => {
    const props = {
      errors: {},
      handleChange: () => {},
      handleSubmit: () => {},
      isLoading: true
    };
    const wrapper = mount(<SignupForm {...props} />);
    console.log(wrapper.prop('isLoading'), '------------------');
    expect(wrapper.prop('isLoading')).toEqual(true);
  });
});
