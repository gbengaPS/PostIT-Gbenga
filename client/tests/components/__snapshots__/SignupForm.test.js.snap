// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component  should render correctly 1`] = `
<div
  className="row"
>
  <Nav />
  <div
    className="col s12 m4 offset-m4"
  >
    <div
      className="form-container"
    >
      <p
        className="center grey-text big"
      >
        Signup
      </p>
      <p
        className="red-text"
      >
          
      </p>
      <form
        action="#"
        method="POST"
        onSubmit={[Function]}
      >
        <InputField
          classnames=""
          handleChange={[Function]}
          labelValue="Full name"
          name="fullName"
          type="text"
        />
        <span
          className="red-text"
        >
           
        </span>
        <InputField
          classnames=""
          handleChange={[Function]}
          labelValue="Username"
          name="username"
          type="text"
        />
        <span
          className="red-text"
        >
           
        </span>
        <InputField
          classnames=""
          handleChange={[Function]}
          labelValue="Email"
          name="email"
          type="email"
        />
        <span
          className="red-text"
        >
           
        </span>
        <InputField
          classnames=""
          errorClass="valid"
          handleChange={[Function]}
          labelValue="Phone Number"
          name="phoneNumber"
          type="number"
        />
        <span
          className="red-text"
        >
           
        </span>
        <InputField
          classnames=""
          errorClass="valid"
          handleChange={[Function]}
          labelValue="Password"
          name="password"
          type="password"
        />
        <span
          className="red-text"
        >
           
        </span>
        <SubmitButton
          disabled=""
          submitValue="Signup"
        />
        <p
          className="center"
        >
           Already have and acount?
          <Link
            replace={false}
            to="login"
          >
             Login 
          </Link>
        </p>
        <p />
      </form>
    </div>
  </div>
</div>
`;
